AccountingSystem/AccountingSystem/GasStation.cpp

```diff
@@ -0,0 +1,92 @@
+#include <sstream>
+#include <string>
+#include <vector>
+#include <ctime>
+#include <iomanip>
+#include "GasStation.h"
+
+GasStation::GasStation(IDataHandler<GasStationData>* handler):_handler(handler)
+{
+}
+
+void GasStation::AddRecord(InputData userData,const bool IncreaseTheBallance)
+{
+    GasStationData data;
+    data.Name = userData.Name;
+    data.Surname = userData.Surname;
+    data.Liters = userData.Liters;
+    data.ID = GenerateNextId();
+    data.Date = GetCurrentDate();
+    auto ballance = GetCurrentBallance();
+    data.Ballance = IncreaseTheBallance ? ballance + data.Liters : ballance - data.Liters;
+	_handler->Add(data);
+}
+
+int GasStation::DeleteRecord(const int id)
+{
+    return _handler->Delete(id);
+}
+
+
+GasStation::GasStationData GasStation::GetRecord(int id)
+{
+    return _handler->Get(id);
+}
+
+std::string GasStation::GetCSVFormat(GasStationData data)
+{
+	std::stringstream ss;
+	const auto delim = ',';
+	ss << data.ID << delim
+		<< data.Name << delim
+		<< data.Surname << delim
+		<< data.Date << delim
+		<< data.Liters << delim
+		<< data.Ballance << delim;
+	return ss.str();
+}
+
+GasStation::GasStationData GasStation::GetGasStationData(std::string line)
+{
+	GasStationData data;
```

Comment:
very optional and style dependent, but many folks suggest to write like this
```suggestion
	auto data = GasStationData{};
```